node-graph/README.md

```diff
@@ -6,20 +6,31 @@ Graphite is an image editor which is centred around a node based editing workflo
 
 ## The Document Graph
 
-The graph that is presented to users in the editor is known as the document graph. Each node that has been placed in this graph has the following properties:
+The graph that is presented to users in the editor is known as the document graph, and is defined in the `NodeNetwork` struct. Each node that has been placed in this graph has the following properties:
 
 ```rs
 pub struct DocumentNode {
-	// An identifier used to display in the editor and to display the appropriate properties.
+	/// An identifier used to display in the UI and to display the appropriate properties.
 	pub name: String,
-	// A NodeInput::Node { node_id, output_index } specifies an input from another node.
-	// A NodeInput::Value { tagged_value, exposed } specifies a constant value. An exposed value is visible as a dot in the node graph UI.
-	// A NodeInput::Network(Type) specifies a node will get its input from outside the graph, which is resolved later.
+	/// The inputs to a node, which are either:
+	/// - From other nodes within this graph [`NodeInput::Node`],
+	/// - A constant value [`NodeInput::Value`],
+	/// - A [`NodeInput::Network`] which specifies that this input is from outside the graph, which is resolved in the graph flattening step.
 	pub inputs: Vec<NodeInput>,
+	/// TODO: what is this?
```

Comment:
Usually compose nodes are automatically inserted to link two nodes together. When the node specifies a manual composition input it opts out of this behaviour and calls the previous node itself

Reply:
(This is a replacement for the short circuit system)

Reply:
That explains why the `manual_composition` in the document network leads to a `ProtoNodeInput::ShortCircut` which is what was confusing to me. Perhaps the document graph concept and proto graph concept should be named the same thing @TrueDoctor?

Reply:
Yeah, that seems reasonable I'm not sure what my reason was for not renaming that as well in my pr

Reply:
ProtoNodeInput::Network is the same as ProtoNodeInput::ManualComposition (formerly ProtoNodeInput::ShortCircuit) so I changed them both to ProtoNodeInput::ManualComposition.

Reply:
Network = ManualComposition sounds strange, are you sure about that?

Reply:
If we remove the Network type we would also need to remove the proto network inputs but I'm not sure that that would be a good idea. Having the inputs allows us to do network specialization without having to recompile the entire network similar to the concept of parameterized shaders

Reply:
There was no difference in the implementation. They both just specify that the protonode's input will be given to it as an argument to `.eval(x)`.

Reply:
I'm not really sure I understand how merging two identical enum variants relates to making a shader parameterised (presumably through uniforms in the graphics api which are used in the shader glsl code).

Reply:
Yes, the idea would be to have the network input as a placeholder which is replaced before the graph execution and actually make it impossible to execute a network with graph inputs present

Reply:
But that is also something that we could add back in the future if that is needed and just remove it for now to make the code less confusing